biometricks/src/main/java/com/handstandsam/biometricks/internal/BiometricksHelper.kt

```diff
@@ -78,4 +86,34 @@ internal class BiometricksHelper(context: Context) {
             }
         }
     }
+
+    /**
+     * Check if the user can securely authenticate with biometrics.
+     */
+    fun canSecurelyAuthenticate(): Boolean {
+        if (Build.VERSION.SDK_INT < 28) {
+            // we only have fingerprint, checking if there are any enrolled fingerprints should be enough.
+            val service = (context.getSystemService(Context.FINGERPRINT_SERVICE) as FingerprintManager?)
+                ?: return false
+            return service.hasEnrolledFingerprints()
+        }
+        try {
+            val keystore = KeyStore.getInstance("AndroidKeyStore")
+            KeyGenerator.getInstance("AES", keystore.provider)
+                    .init(
```

Comment:
We tried a similar approach and `init` returns without an exception on some devices/OSs even when no fingerprints are enrolled. e.g. LG G6 running Android 7. 

So I suggest doing something like `keyGeneratorInitDoesntThrow() && biometricManager.canAuthenticate() == BIOMETRIC_SUCCESS`

Reply:
good to know!

Reply:
Interesting.  Thanks.